### PR TITLE
Adding ServerInput class functionality using PortableRest

### PR DESCRIPTION
**Pull request by arunkt1**
I have to make further tests to ascertain whether response.Result returns the deserialized string representation of the device state from the server which can be directly passed to the updateDevice() function

### CODE REVIEW

File: api/DeviceAPI.cs [hunk 1]
```diff
@@ -30,11 +30,13 @@ public Interfaces(Uri serverAddress)
 	 * \param[in] address Location of the house to query for unregistered devices.
 	 * \param[out] List of strings which represent devices which could be registered.
 	 */
-	public List<string> enumerateDevices(UInt64 house_id)
+
+
+	public bool enumerateDevices(UInt64 house_id)
```
Comment (jasedit) [inline api/DeviceAPI.cs:35]: This needs to return a List of strings which identify the various devices.

File: api/ServerInput.cs [hunk 2]
```diff
@@ -1,23 +1,80 @@
...
+		{
+		var client = new RestClient();
+		client.BaseUrl = _server_uri;
+		var request = new RestRequest("api/storage/device/{house}/{space}/{device}");//the default request is GET
+		request.AddUrlSegment("house", dev.ID.HouseID.ToString());
+		request.AddUrlSegment("space", dev.ID.HouseID.ToString());
+		request.AddUrlSegment("device", dev.ID.HouseID.ToString());
+		var response = client.ExecuteAsync<string>(request);//tries to get a deserialized string of device state as the response.
+		 
+			if(response.Status ==TaskStatus.Created)//condition check for a successful completion of the GET request
```
Comment (jasedit) [inline api/ServerInput.cs:44]: Is this the right code? Not an OK or anything else? I haven't double checked the TaskStatus values, but this seems counter-intuitive given that this is a GET request.

File: api/ServerInput.cs [hunk 3]
```diff
@@ -1,23 +1,80 @@
...
+			{	
+			bool update_status = Interfaces.UpdateDevice(dev, response.Result);// updating the device state obtained from server
+															//the assumption is that response.Result has the deserialized device state
+			if(update_status == true)
+				{
+				server_input_status = true;
+			
+				}
```
Comment (jasedit) [inline api/ServerInput.cs:52]: Format hammer needed.

File: api/ServerInput.cs [hunk 4]
```diff
@@ -1,23 +1,80 @@
...
+		//RestResponse response; //= client.ExecuteAsync(request);
+		//RestResponse response = client.ExecuteAsync(request);
+		//Assert.AreEqual(HttpStatusCode.OK, resp.StatusCode);
+		//Device dev1= response.Result;
+		//dev = dev1;
+		//var content = response.Result;
+
+		//Device DeserializeDevice(string info, IDeviceInput inp, IDeviceOutput outp, TimeFrame frame)
+		//POST api/house/device/state
+		//return true;
```
Comment (jasedit) [inline api/ServerInput.cs:77]: You can likely nuke the scar tissue.

File: api/ServerInput.cs [hunk 5]
```diff
@@ -1,23 +1,80 @@
-﻿/**
+﻿	/**
```
Comment (jasedit) [inline api/ServerInput.cs:1]: Remove the leading whitespace here.

**Comment by arunkt1** [comment]
Thank you Jason. I haven't really started to modify enumeratedevices(). I will work on that and confirm the working of PortableRest by today.

**Comment by jasedit** [comment]
Closed for PortableRest crashing iOS.

**Pull request closed without merging**